### PR TITLE
Add guide to run control-plane locally by using envoy

### PR DESCRIPTION
**Pull request by tnqv**
**What this PR does / why we need it**:
- I followed old guide from [Control-plane-development document](https://pipecd.dev/docs/contribution-guidelines/control-plane-development/), but I failed to run the application correctly ( which I already reported it [here](https://github.com/pipe-cd/pipe/issues/1049#issuecomment-721594950) ).

- I successfully run it with Envoy which based on configuration from [this](https://github.com/pipe-cd/pipe/blob/master/manifests/pipecd/templates/envoy-configmap.yaml).

- I want to add up some notes about installing Envoy step. If is there any redundant configs, please let me know. Thank you! 

**Which issue(s) this PR fixes**:
#1081 

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/content/en/docs/contribution-guidelines/control-plane-development.md [hunk 1]
```diff
@@ -24,4 +26,146 @@ bazelisk run //cmd/server:server -- server \
...
+      load_assignment:
+        cluster_name: server-http
+        endpoints:
+        - lb_endpoints:
+          - endpoint:
+              address:
+                socket_address:
+                  address: localhost
+                  port_value: 9082
+```
```
Comment (nghialv) [inline docs/content/en/docs/contribution-guidelines/control-plane-development.md:141]: nit: Maybe we should add a `local` directory at the root of the repository to store those files (`envoy-config.yaml`, `Dockerfile`, `piped-config.yaml`).

File: docs/content/en/docs/contribution-guidelines/control-plane-development.md [hunk 2]
```diff
@@ -24,4 +26,146 @@ bazelisk run //cmd/server:server -- server \
...
+
+#### Using Envoy running on Docker
+
+1. Create simple Dockerfile.
+
+```Dockerfile
+FROM envoyproxy/envoy:v1.10.0
+COPY absolute-path-to-pipe-envoy-config.yaml /etc/envoy/envoy.yaml
+CMD /usr/local/bin/envoy -c /etc/envoy/envoy.yaml
+```
```
Comment (nghialv) [inline docs/content/en/docs/contribution-guidelines/control-plane-development.md:157]: nit: The path of the envoy-config.yaml can be a relative one.

**Comment by nghialv** [comment]
Hi @tnqv Thank you for your great PR, as always.
I have just left some nits.

**Comment by tnqv** [comment]
@nghialv Thanks for your reviewing.
I created one `local` folder and add `Dockerfile` & `envoy-config.yaml`. Also change the guide documents.
Please check it again when you have time!
Thank you!

**Comment by nghialv** [comment]
@tnqv Nice. Thank you very much.
/approve

**Pull request merged**